src/dune_rules/melange/melange_rules.ml

```diff
@@ -195,17 +214,24 @@ let build_js
       in
       With_targets.map_build command ~f:(fun command ->
         let open Action_builder.O in
-        let local_library_paths =
-          match local_modules with
-          | Some (modules, obj_dir) ->
+        match local_modules_and_obj_dir with
+        | Some (modules, obj_dir) ->
+          let paths =
             let+ module_deps =
               Dep_rules.immediate_deps_of m modules ~obj_dir ~ml_kind:Impl
             in
-            List.map module_deps ~f:(fun dep_m ->
-              Obj_dir.Module.cm_file_exn obj_dir dep_m ~kind:(Melange Cmj) |> Path.build)
-          | None -> Action_builder.return []
-        in
-        Action_builder.dyn_paths_unit local_library_paths >>> command)
+            List.fold_left module_deps ~init:[] ~f:(fun acc dep_m ->
+              if Module.has dep_m ~ml_kind:Impl
```

Comment:
Why is this check necessary if `immediate_deps_of` is being called with `~ml_kind:Impl`?

Reply:
Because the immediate dependencies could be intf-only modules which don’t have cmj